### PR TITLE
Use os-agnostic service helpers

### PR DESCRIPTION
**Pull request by ekohl**
Maybe we should also include os-agnostic tServiceStart and tServiceEnable.

### CODE REVIEW

**Comment by stbenjam** [comment]
> Maybe we should also include os-agnostic tServiceStart and tServiceEnable.

:+1:

**Comment by ekohl** [comment]
See https://github.com/theforeman/foreman-bats/pull/103 for an initial implementation. If that's merged then I'll rebase this.

File: bats/os_helper.bash [hunk 1]
```diff
@@ -122,3 +124,23 @@ tCommandExists() {
 tFileExists() {
   [[ -f "$1" ]]
 }
+
+tRHSubscribeAttach() {
```
Comment (stbenjam) [inline bats/os_helper.bash]: Where's this used?

Comment (ekohl) [inline bats/os_helper.bash]: Nowhere yet, I just copied os_helper.bash from foreman-bats which includes this.

**Comment by ekohl** [comment]
Rewritten since https://github.com/Katello/katello-deploy/pull/228 already simplified the code it was fixing. It's now aimed at using OS-agnostic service helpers.

I haven't had time yet to test this, so consider it RFC.

**Comment by stbenjam** [comment]
@ekohl This still looks like a wortwhile refactor, if you want to rebase, I can review/test.

**Comment by ekohl** [comment]
Rebased, but untested

**Comment by ekohl** [comment]
FYI: the os_helper is just a copy from foreman-bats with ```tNonZeroFile``` re-added.

**Comment by ehelms** [comment]
@stbenjam please review when you get a chance

**Comment by stbenjam** [comment]
Sorry for the delay, this looks OK to me.

**Pull request merged**